src/main/java/org/apache/commons/lang3/concurrent/AtomicSafeInitializer.java

```diff
@@ -53,12 +53,13 @@
  */
 public abstract class AtomicSafeInitializer<T> extends AbstractConcurrentInitializer<T, RuntimeException> {
 
+    private final T NO_INIT = (T) new Object(){};
```

Comment:
- Should be `static`, we don't need a new `Object` instance for each new initializer.
- Why is the Object an anonymous class? IOW: `new Object(){};` --> `new Object();`

Reply:
Good call on static.

Object is an anonymous class because that way it will show up as `AtomicSafeInitializer$1` instead of `Object` in a debugger, which makes it easier to distinguish between `NO_INIT` and a return value of type `Object` when debugging.

Reply:
WRT `new Object(){};` vs `new Object();`, IMO, that's not enough of a reason to force an anonymous subclass into the runtime. It's also different from what `LazyInitializer` does, which is confusing to the maintainer at large (especially when uncommented). I'll approve the PR but use `new Object();`.